ui/selectors/selectors.js

```diff
@@ -2982,7 +2984,23 @@ export function getMetaMetricsDataDeletionStatus(state) {
   return state.metamask.metaMetricsDataDeletionStatus;
 }
 
+/**
+ * Gets the remote feature flags from either the manifest or state.
+ * First checks if remote feature flags exist in the manifest and returns those if present.
+ * Otherwise returns the remote feature flags from the MetaMask state that's retrieved from controller.
+ *
+ * @param {object} state - The MetaMask state object
+ * @returns {object} The remote feature flags object containing feature flag key-value pairs
+ */
 export function getRemoteFeatureFlags(state) {
+  const remoteFeatureFlagsValueFromManifest =
+    getManifestFlags().remoteFeatureFlags;
+  if (
+    remoteFeatureFlagsValueFromManifest &&
+    Object.keys(remoteFeatureFlagsValueFromManifest).length > 0
+  ) {
+    return remoteFeatureFlagsValueFromManifest;
+  }
```

Comment:
Were there any discussions about if this should be merged with `state.metamask.remoteFeatureFlags` or completely overridden (like it is now)?

Reply:
There was no discussion, but you made an excellent point that this should be shallow merged.  Modified caa4b4f9780a94e51c494f08800c7ffc3d2ff981